packages/dashboard/src/components/iot-annotations/annotations.ts

```diff
@@ -0,0 +1,71 @@
+import { XAnnotation } from '@synchro-charts/core';
+import { DashboardConfiguration } from '../../types';
+
+export const addXAnnotation = ({
+  dashboardConfiguration,
+  widgetId,
+  annotation,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgetId: string;
+  annotation: XAnnotation;
+}) => {
+  return dashboardConfiguration.widgets.map((widget) => {
```

Comment:
rather than making an action per field, for this field we can just utilize 

```https://github.com/awslabs/iot-app-kit/blob/feature-dashboard/packages/dashboard/src/dashboard-actions/actions.ts#L132-L135```

and skip the action boilerplate